airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeGcsStreamCopier.java

```diff
@@ -30,20 +41,49 @@ public SnowflakeGcsStreamCopier(final String stagingFolder,
     this.filenameGenerator = stagingFilenameGenerator;
   }
 
+  @Override
+  public void copyStagingFileToTemporaryTable() throws Exception {
+    List<List<String>> partition = Lists.partition(new ArrayList<>(gcsStagingFiles), MAX_CHUNK_SIZE);
+    for (int i = 0; i < partition.size(); i++) {
+      LOGGER.info("Starting copy chunk {} to tmp table: {} in destination for stream: {}, schema: {}. Chunks count {}", i, tmpTableName, streamName,
+          schemaName, partition.size());
+      executeCopy(partition.get(i));
+    }
+    LOGGER.info("Copy to tmp table {} in destination for stream {} complete.", tmpTableName, streamName);
+  }
+
+  private void executeCopy(List<String> files) {
+
+    final var copyQuery = String.format(
+        "COPY INTO %s.%s FROM '%s' storage_integration = gcs_airbyte_integration "
+            + " file_format = (type = csv field_delimiter = ',' skip_header = 0 FIELD_OPTIONALLY_ENCLOSED_BY = '\"') "
+            + "files = (" + generateFilesList(files) + " );",
+        schemaName,
+        tmpTableName,
+        generateBucketPath());
+
+    Exceptions.toRuntime(() -> db.execute(copyQuery));
+  }
+
+  public String generateBucketPath() {
+    return "gcs://" + gcsConfig.getBucketName() + "/" + stagingFolder + "/" + schemaName + "/";
+  }
+
+  public String generateFilesList(List<String> files) {
+    StringJoiner joiner = new StringJoiner(",");
+    files.forEach(filename -> joiner.add("'" + filename.substring(filename.lastIndexOf("/") + 1) + "'"));
```

Comment:
do we have any safeguards against `filename` containing quote characters? I.e. is it impossible for `filename` to be something like `oh no ' now we have a syntax error`?

Reply:
from liren, I think by the time we get to this point the filename has already been sanitized. so this should be fine.